### PR TITLE
NEO-2385 FIX: menu not accessible by keyboard inside Table Header

### PR DESCRIPTION
**Pull request by yangAtSpoken**
[Table](https://deploy-preview-547--neo-react-library-storybook.netlify.app/?path=/story/components-table--advanced-filtering-and-sorting)
[Drawer](https://deploy-preview-547--neo-react-library-storybook.netlify.app/?path=/story/components-drawer--basic-drawer)
[Link to original PR where these problems were reported by @Justin-Avaya ](https://github.com/avaya-dux/neo-react-library/pull/530#issuecomment-2344650000)
**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [ ] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

`@avaya-dux/dux-design`: Fix the two accessibility issues reported in previous pr;
- Menus on table header do not open with keyboard
- Can not reach the bottom menu item with keyboard

`@avaya-dux/dux-devs`: 
- Allow menu items wrapped by React Fragment
- Tolerate Tooltip wrapping MenuButton

## Summary by CodeRabbit

- **New Features**
	- Updated the `Drawer` component for improved visibility and animation handling, ensuring a smoother user experience.
  
- **Bug Fixes**
	- Enhanced the robustness of tests for the `Drawer` and `Table` components to better reflect UI state changes.

- **Tests**
	- Updated test cases for the `Drawer` component to validate visibility states comprehensively, utilizing asynchronous checks for accuracy.

### CODE REVIEW

File: src/components/Menu/helpers/helpers.test.jsx [hunk 1]
```diff
@@ -8,15 +8,145 @@ import { SubMenu } from "../SubMenu";
...
+					<MenuItem key="2-2" id="2-2">
+						Edit
+					</MenuItem>
+					<MenuItem key="2-3" id="2-3">
+						Delete
+					</MenuItem>
+				</>,
+			];
+			expect(flattenChildren(children).props.children.length).toEqual(4);
+			expect(flattenChildren(children)).toMatchInlineSnapshot(`
```
Comment (joe-s-avaya) [inline src/components/Menu/helpers/helpers.test.jsx:93]: `toMatchInlineSnapshot` is finicky, but it make sense for this particular issue

File: src/components/Menu/helpers/helpers.test.jsx [hunk 2]
```diff
@@ -8,15 +8,145 @@ import { SubMenu } from "../SubMenu";
...
+				  </MenuItem>
+				  <MenuItem
+				    id="2-3"
+				  >
+				    Delete
+				  </MenuItem>
+				</React.Fragment>
+			`);
+		});
+		it("should not flatten SubMenu", () => {
```
Comment (joe-s-avaya) [inline src/components/Menu/helpers/helpers.test.jsx:118]: great test 👍

**Review by joe-s-avaya — APPROVED** [review]

**Comment by Justin-Avaya** [comment]
Functionally, this update solves the initial two PR items

**Design/UX** 
@matthewAvaya - When opening a filter menu that is in a default state (no previous sorting) the first menu item is "Clear Sort." This item is disabled, in focus and allows me to select it by mouse and keyboard. Doing so however, does nothing and dismisses the menu which is annoying. I'm having a hard time finding a 1:1 reference for this pattern and I believe it needs further discussion. 

As a user, I would expect: 
- This row would not be in focus and/or selectable 
-  - Rational : I am used to the menu dropdown experience where if a menu row item is in a disabled state…
- - - Clicking on the row with a mouse does nothing and the menu is not dismissed 
- - - When navigating the filter using a keyboard, the navigation skips over any disabled/unselectable row.

Tagging everyone as I am not sure if this is out of scope and needs further discussion if it should pertain to this PR or not.
If out of scope, I will approve the PR as it indeed solves the two initial issues.
@joe-s-avaya 
@enrique-prado 

**Example 01**


**Example 02**

**Comment by Justin-Avaya** [comment]
@yangAtSpoken 
Looking at the Figma design, it does not appear we have "clear sort" as option for filtering.
Screenshot 2024-09-24 at 12 05 07 PM

**Comment by matthewAvaya** [comment]
> Functionally, this update solves the initial two PR items
> 
> **Design/UX** @matthewAvaya - When opening a filter menu that is in a default state (no previous sorting) the first menu item is "Clear Sort." This item is disabled, in focus and allows me to select it by mouse and keyboard. Doing so however, does nothing and dismisses the menu which is annoying. I'm having a hard time finding a 1:1 reference for this pattern and I believe it needs further discussion.
> 
> As a user, I would expect:
> 
> * This row would not be in focus and/or selectable
> * * Rational : I am used to the menu dropdown experience where if a menu row item is in a disabled state…
> * * * Clicking on the row with a mouse does nothing and the menu is not dismissed
> * * * When navigating the filter using a keyboard, the navigation skips over any disabled/unselectable row.
> 
> Tagging everyone as I am not sure if this is out of scope and needs further discussion if it should pertain to this PR or not. If out of scope, I will approve the PR as it indeed solves the two initial issues. @joe-s-avaya @enrique-prado
> 
> **Example 01**  [  ](https://private-user-images.githubusercontent.com/161497909/370401627-6031a27c-5e0b-4502-9c70-46b90bb4d2ca.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcyMDQ3NDYsIm5iZiI6MTcyNzIwNDQ0NiwicGF0aCI6Ii8xNjE0OTc5MDkvMzcwNDAxNjI3LTYwMzFhMjdjLTVlMGItNDUwMi05YzcwLTQ2YjkwYmI0ZDJjYS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyNFQxOTAwNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YTUyMWYyZDk2NWVkMjM0MGFiYTVhOWEzZmVkNjU5ODliMWZjODA1MjZhMjE3Njk1NzVjYmJjM2Q5ZTMzNzQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.wR27VAFU5sc2ZRaagPz943jGnGoavDPPKSxeICzLsWA) [ ](https://private-user-images.githubusercontent.com/161497909/370401627-6031a27c-5e0b-4502-9c70-46b90bb4d2ca.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcyMDQ3NDYsIm5iZiI6MTcyNzIwNDQ0NiwicGF0aCI6Ii8xNjE0OTc5MDkvMzcwNDAxNjI3LTYwMzFhMjdjLTVlMGItNDUwMi05YzcwLTQ2YjkwYmI0ZDJjYS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyNFQxOTAwNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YTUyMWYyZDk2NWVkMjM0MGFiYTVhOWEzZmVkNjU5ODliMWZjODA1MjZhMjE3Njk1NzVjYmJjM2Q5ZTMzNzQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.wR27VAFU5sc2ZRaagPz943jGnGoavDPPKSxeICzLsWA)
> 
> **Example 02**  [  ](https://private-user-images.githubusercontent.com/161497909/370402187-2af14076-fff8-4ece-af9d-d668e8e72c86.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcyMDQ3NDYsIm5iZiI6MTcyNzIwNDQ0NiwicGF0aCI6Ii8xNjE0OTc5MDkvMzcwNDAyMTg3LTJhZjE0MDc2LWZmZjgtNGVjZS1hZjlkLWQ2NjhlOGU3MmM4Ni5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyNFQxOTAwNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yOTE0OTJhOWE2ODk3MjBiNTZjOTk1ZjY1NmY1ZGE0ZjU3ODBlOWEwNmE0ZDMzYmUxZGUzZGJjZmVhODY1YmUxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.NC17ZWb0zYOq5hBgvDCSJ-Kut_ZxycT1frpSy6rnmSo) [ ](https://private-user-images.githubusercontent.com/161497909/370402187-2af14076-fff8-4ece-af9d-d668e8e72c86.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcyMDQ3NDYsIm5iZiI6MTcyNzIwNDQ0NiwicGF0aCI6Ii8xNjE0OTc5MDkvMzcwNDAyMTg3LTJhZjE0MDc2LWZmZjgtNGVjZS1hZjlkLWQ2NjhlOGU3MmM4Ni5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyNFQxOTAwNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yOTE0OTJhOWE2ODk3MjBiNTZjOTk1ZjY1NmY1ZGE0ZjU3ODBlOWEwNmE0ZDMzYmUxZGUzZGJjZmVhODY1YmUxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.NC17ZWb0zYOq5hBgvDCSJ-Kut_ZxycT1frpSy6rnmSo)

There should be no 'Clear Sort' action on the filter. I'm not sure why it's there. I thought we had discussed this in the past, but can't remember to who. Suggest removing this action under all column headers.

**Comment by joe-s-avaya** [comment]
> There should be no 'Clear Sort' action on the filter. I'm not sure why it's there. I thought we had discussed this in the past, but can't remember to who. Suggest removing this action under all column headers.

@matthewAvaya I could have sworn that we did want a "clear" option. It's weird to me that you wouldn't allow someone deselect.

**Comment by matthewAvaya** [comment]
> > There should be no 'Clear Sort' action on the filter. I'm not sure why it's there. I thought we had discussed this in the past, but can't remember to who. Suggest removing this action under all column headers.
> 
> @matthewAvaya I could have sworn that we did want a "clear" option. It's weird to me that you wouldn't allow someone deselect.

I don't know if it's needed. I'm looking around, specifically Microsoft, and they don't include this action. I would guess by default, tables are ordered from newest to oldest. If so, they can use the Date column to view it in that order, if it's included. Otherwise, the user will select their preferred order by selecting a specific column to sort. I would vote to remove it.

File: src/components/Drawer/Drawer.tsx [hunk 3]
```diff
@@ -151,6 +152,26 @@ const BasicDrawer = ({
...
+			setTimeout(() => {
+				drawer.classList.add("neo-drawer--open");
+			}, 10);
+		} else {
+			drawer.classList.remove("neo-drawer--open");
+			setTimeout(() => {
+				drawer.classList.remove("neo-drawer--visible");
+			}, 300);
+		}
+	}, [open]);
```
[the remark this answers is not in the document]
Comment (yangAtSpoken) [reply src/components/Drawer/Drawer.tsx]: Applied timer cleanup logic.

**Comment by yangAtSpoken** [comment]
@Justin-Avaya and @matthewAvaya 
Created [neo-2415](https://avaya.atlassian.net/browse/NEO-2415) to change disabled MenuItem behavior
Created [neo-2416](https://avaya.atlassian.net/browse/NEO-2416) to remove "Clear Sort"

**Comment by yangAtSpoken** [comment]
@enrique-prado I fixed Drawer tab issue and kept the transition. Please review. @joe-s-avaya Please look again. Thx.

File: src/components/Drawer/Drawer_shim.css [hunk 4]
```diff
@@ -6,18 +6,22 @@
...
 	background-color: var(--sheet-background);
 }
 
-.neo-drawer--isOpen {
+.neo-drawer--open {
+	display: flex;
 	transform: translate(0);
 	box-shadow: 0px 8px 16px var(--sheet-shadow);
+
+	@starting-style {
```
Comment (joe-s-avaya) [inline src/components/Drawer/Drawer_shim.css:22]: oooh, I didn't know this was a thing, nice work!

**Comment by joe-s-avaya** [comment]
> @enrique-prado I fixed Drawer tab issue and kept the transition. Please review. @joe-s-avaya Please look again. Thx.

@yangAtSpoken looks good! Nice work with the `@starting-style`, I learned something new!

**Comment by matthewAvaya** [comment]
It's looking good, and thanks for fixing the tab order. I'm seeing one issue. When I select the filter action from any of the columns besides Name, and select Cancel or Apply, the focus goes back to the first column Name. I would expect that the focus would go back to the column that I was where the filter action was done.

**Review by enrique-prado — APPROVED** [review]
Nice work.

**Review by matthewAvaya — APPROVED** [review]

**Review by Justin-Avaya — APPROVED** [review]

**Comment by yangAtSpoken** [comment]
> It's looking good, and thanks for fixing the tab order. I'm seeing one issue. When I select the filter action from any of the columns besides Name, and select Cancel or Apply, the focus goes back to the first column Name. I would expect that the focus would go back to the column that I was on where the original filter action was done.

@matthewAvaya Created [this story](https://avaya.atlassian.net/browse/NEO-2417) from your comment.

**Pull request merged**